OneSignal/OneSignal-iOS-SDK
# [JWT] Handle logout when Identity verification is on

# Description
## One Line Summary
Handle logout when JWT is on by disabling the push subscription and still create an anonymous local user to prevent data updates.

## Details

### Motivation
When JWT is on, logout needs to be handled differently. This is a small PR, so logout is handled simply.

### Scope
- Logout will mostly be the same: an anonymous user is created locally but user creation will be blocked from sending to the server and this anonymous user's requests are dropped. Changing users in the SDK prevents mixing up data or making updates to the old user incorrectly.
- Additionally, disable push sub when logout called with JWT on by using a local flag, which will send a subscription update request.
- When login is next called, this flag will revert and the push subscription will maintain its original state.

# Testing
## Unit testing
❗️TODO

## Manual testing
To fill out
Tested on iPhone 13 on iOS 17 
1. Be logged in and push permission granted
2. Call logout and see the push subscription is disabled on server
3. Kill and re-open app to drive a new session
4. No IAM is fetched due to no viable alias
5. Log back into the same user and IAM is now fetched (driven by user change)
6. The login generates a user create request that sends the correct push subscription state from before disabling

# Affected code checklist

# Checklist
## Overview

## Testing

## Final pass


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/OneSignal/OneSignal-iOS-SDK/1488)
